src/compiler/binder.ts

```diff
@@ -3948,6 +3948,9 @@ export function getContainerFlags(node: Node): ContainerFlags {
         case SyntaxKind.ClassStaticBlockDeclaration:
             return ContainerFlags.IsContainer | ContainerFlags.IsControlFlowContainer | ContainerFlags.HasLocals | ContainerFlags.IsFunctionLike;
 
+        case SyntaxKind.JSDocImportTag:
+            return ContainerFlags.IsContainer | ContainerFlags.IsControlFlowContainer | ContainerFlags.HasLocals;

```

Comment:
Oh wow, this seems good

Reply:
As I understand it, these 3 flags are to prevent mistakenly using a function host as the container for the imported bindings. Is that correct? Can you add a comment like that? I didn't naturally expect imports to be containers otherwise.

(This won't be a problem in Corsa since tags that generate statements won't even have hosts for the generated statements.)

Reply:
I’ve added the comment